### PR TITLE
Move web integration tool tests to web.shard

### PR DESCRIPTION
**Pull request by annagrin**
## Description

Web integration tool tests depend on DDC changes in SDK. This change
moves them to a separate shard and sub shard so CI bot configurations
can run them separately.

In particular, this will allow running those tests on dart CI `flutter HHH web bot` 
instead of a non-web one, allowing early detection and easy classification of 
issues caused by SDK changes as VM- or Web related.

## Related Issues

Related: https://github.com/flutter/flutter/issues/69711
Closes: https://github.com/flutter/flutter/pull/69914

## Tests

This change is only in tests.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/logger.dart [hunk 1]
```diff
@@ -334,7 +334,7 @@ class StdoutLogger extends Logger {
   void writeToStdErr(String message) => _stdio.stderrWrite(message);
 
   @override
-  void printTrace(String message) { }
+  void printTrace(String message) { }// => _stdio.stdoutWrite('TRACE: $message');
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/base/logger.dart:337]: nit: remove comment

File: dev/prod_builders.json [hunk 2]
```diff
@@ -114,6 +114,12 @@
          "task_name": "linux_tool_tests",
          "flaky": false
       },
+      {
```
Comment (jonahwilliams) [inline dev/prod_builders.json:117]: I don't think mac has chrome configured correctly for the web tests

**Review by jonahwilliams — APPROVED** [review]
Nice!


LGTM besides nits, though @godofredoc for order that is safe to land in

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 1 above)
Comment (annagrin) [reply packages/flutter_tools/lib/src/base/logger.dart:337]: thanks!

**Comment by jonahwilliams** [comment]
It would actually be nice if we could configure the web tests to run at a higher verbosity and print everything for debugging - since we know they are flaky

**Comment by annagrin** [comment]
Following recommendation from @godofredoc - removed the bot configuration changes, as they need to be submitted after the infra changes are in.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
Making the call that this is not affecting google3, landing so we can get these tests running again soon

**Pull request merged**